### PR TITLE
bugfix to make dir.chdir threadsafe

### PR DESCRIPTION
**Pull request by stevenoneill**
### Description

Makes Dir.chdir threadsafe.

### Issues Resolved

Resolves https://github.com/chef/mixlib-archive/issues/28

Considered:
* https://github.com/test-kitchen/test-kitchen/issues/1416
* https://github.com/ruby-git/ruby-git/pull/372

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by stevenoneill** [comment]
Signed-off-by: Steven ONeill oneill.steven@gmail.com

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 # Change Log
 
-<!-- latest_release 0.4.11 -->
-## [v0.4.11](https://github.com/chef/mixlib-archive/tree/v0.4.11) (2018-07-31)
+<!-- latest_release 0.4.12 -->
+## [v0.4.12](https://github.com/chef/mixlib-archive/tree/v0.4.12) (2018-08-01)
 
 #### Merged Pull Requests
-- Add github issue template &amp; standardize expeditor config [#26](https://github.com/chef/mixlib-archive/pull/26) ([tas50](https://github.com/tas50))
+- Update codeowners and add PR template [#27](https://github.com/chef/mixlib-archive/pull/27) ([tas50](https://github.com/tas50))
```
Comment (cheeseplus) [inline CHANGELOG.md]: You shouldn't update the changelog, that's handled automatically by Expeditor and it's not good practice to increment the version in a PR (that's for the maintainers to do)

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-0.4.11
\ No newline at end of file
+0.4.12
```
Comment (cheeseplus) [inline VERSION]: same as above

**Comment by coderanger** [comment]
While this might be a decent stopgap, the correct solution is that mixlib-archive shouldn't be using chdir at all.

File: VERSION (hunk 2 above)
Comment (stevenoneill) [reply VERSION]: boo-boo on my part....will get it right

**Comment by stevenoneill** [comment]
Hmmm....did I mess something up here?  (first ever PR)

**Comment by tas50** [comment]
Thanks @stevenoneill we'll get this pushed up asap

**Comment by stevenoneill** [comment]
Awesome....thank you all!  Really appreciate for the quick attention to this!

**Pull request merged**